src/net.cpp

```diff
@@ -1454,7 +1454,11 @@ void ThreadDNSAddressSeed()
                     found++;
                 }
             }
-            addrman.Add(vAdd, CNetAddr(seed.name, true));
+
+            // TODO: It doesn't make sense to define an arbitrary IP as the source of others.
+            // Instead, a dummy range should be created to identify seed resolves.
+            if (!vIPs.empty())
+                addrman.Add(vAdd, vIPs[0]);
```

Comment:
That's not actually correct. The "source IP" the addresses get assigned to should be stable over multiple runs, as the DNS seed is seen as a potential attacker who is only allowed to modify a certain subset of entries. If the IP changes over time, that protection disappears.